recipes/tests/test_models.py

```diff
@@ -18,3 +26,54 @@ def test_str(self):
         expected = 'test_title'
         actual = Recipe.objects.get(title='test_title').__str__()
         self.assertEqual(expected, actual)
+
+    def test_upvote_count(self):
+        """
+        Validate total number of upvotes for a recipe
+        """
+        expected = 2
+        actual = self.recipe1.vote_objects.filter(up=True).count()
+        self.assertEqual(expected, actual)
+
+    def test_vote_objects(self):
+        """
+        Validate total number of votes for a recipe
+        """
+        expected = 3
+        actual = len(self.recipe1.vote_objects)
+        self.assertEqual(expected, actual)
+
+    def test_cast_vote_none(self):
+        """
+        Change a user vote to None
+        """
+        self.recipe1.cast_vote(user=self.user1, up=None)
+        expected = 1
+        actual = self.recipe1.vote_objects.filter(up=True).count()
+        self.assertEqual(expected, actual)
+
+    def test_cast_vote_false(self):
```

Comment:
good tests.

This is a personal preference, but I'll use double underscores or "dunders" when I am testing the same _thing_ multiple times with different conditions. Like you have `test_cast_vote_false`, I would use the name `test_cast_vote__false` and `test_cast_vote__true`. It groups the tests together, and allows you to quickly see _just what's different_ in the conditions of the test.

Awesome use of docstrings here too though

Great job

Reply:
Added another dunder for the three test cases